tls/extensions/s2n_npn.c

```diff
@@ -24,7 +24,13 @@
 
 bool s2n_npn_should_send(struct s2n_connection *conn)
 {
-    return s2n_client_alpn_should_send(conn) && conn->config->npn_supported;
+    /*
+     *= https://datatracker.ietf.org/doc/id/draft-agl-tls-nextprotoneg-04#section-3
+     *# For the same reasons, after a handshake has been performed for a
+     *# given connection, renegotiations on the same connection MUST NOT
+     *# include the "next_protocol_negotiation" extension.
+     */
+    return s2n_client_alpn_should_send(conn) && conn->config->npn_supported && !s2n_handshake_is_renegotiation(conn);
```

Comment:
Is this as simple as just not sending the extension? The "For the same reasons" refers to:
>  Unlike many other TLS extensions, this extension does not establish
   properties of the session, only of the connection.  When session
   resumption or session tickets [RFC5077] are used, the previous
   contents of this extension are irrelevant and only the values in the
   new handshake messages are considered.

That might imply we should keep the same protocol for the renegotiated handshake, not just have no protocol. Or maybe we're fine assuming that the peer has already called get_protocol for the first handshake, and won't call it again for the second?

Reply:
Like, do you think you need to preserve the protocol during `s2n_renegotiate_wipe` if npn was used?

Reply:
Yeah I mean whether or not we preserve this protocol through the wipe is a matter of opinion. 🤷‍♀️ I mean, do I want to write complex preservation logic for only the NPN protocol? No. Is that the nicest thing to do for users? Probably 🙄.